### ISSUE TITLE
`phab -cddev install --force` not working with npm

### ISSUE THREAD

**Issue by mgrsskls** [factorial-io/phabalicious#351]
When switching from yarn to npm in our project, we get the following error when running `phab -cddev install --force`:

```
PHP Fatal error:  Uncaught TypeError: Phabalicious\Method\RunCommandBaseMethod::runCommand(): Argument #3 ($command) must be of type string, null given, called in phar:///usr/local/bin/phab/src/Method/NpmMethod.php on line 58 and defined in phar:///usr/local/bin/phab/src/Method/RunCommandBaseMethod.php:148
Stack trace:
#0 phar:///usr/local/bin/phab/src/Method/NpmMethod.php(58): Phabalicious\Method\RunCommandBaseMethod->runCommand(Object(Phabalicious\Configuration\HostConfig), Object(Phabalicious\Method\TaskContext), NULL)
#1 phar:///usr/local/bin/phab/src/Method/NpmMethod.php(63): Phabalicious\Method\NpmMethod->resetPrepare(Object(Phabalicious\Configuration\HostConfig), Object(Phabalicious\Method\TaskContext))
#2 phar:///usr/local/bin/phab/src/Method/MethodFactory.php(196): Phabalicious\Method\NpmMethod->installPrepare(Object(Phabalicious\Configuration\HostConfig), Object(Phabalicious\Method\TaskContext))
#3 phar:///usr/local/bin/phab/src/Method/MethodFactory.php(150): Phabalicious\Method\MethodFactory->callImpl(Object(Phabalicious\Method\NpmMethod), 'installPrepare', Object(Phabalicious\Configuration\HostConfig), Object(Phabalicious\Method\TaskContext), true)
#4 phar:///usr/local/bin/phab/src/Method/MethodFactory.php(107): Phabalicious\Method\MethodFactory->runTaskImpl('installPrepare', Object(Phabalicious\Configuration\HostConfig), Object(Phabalicious\Method\TaskContext), false)
#5 phar:///usr/local/bin/phab/src/Command/InstallCommand.php(78): Phabalicious\Method\MethodFactory->runTask('install', Object(Phabalicious\Configuration\HostConfig), Object(Phabalicious\Method\TaskContext), Array)
#6 phar:///usr/local/bin/phab/vendor/symfony/console/Command/Command.php(255): Phabalicious\Command\InstallCommand->execute(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#7 phar:///usr/local/bin/phab/vendor/symfony/console/Application.php(1039): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#8 phar:///usr/local/bin/phab/vendor/symfony/console/Application.php(275): Symfony\Component\Console\Application->doRunCommand(Object(Phabalicious\Command\InstallCommand), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#9 phar:///usr/local/bin/phab/vendor/symfony/console/Application.php(149): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#10 phar:///usr/local/bin/phab/bin/phab(65): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#11 /usr/local/bin/phab(14): require('phar:///usr/loc...')
#12 {main}
  thrown in phar:///usr/local/bin/phab/src/Method/RunCommandBaseMethod.php on line 148
```

Some input by @kleinfreund:

> I took a random guess and thought to compare the recent commit history for
>
>- https://github.com/factorial-io/phabalicious/commits/66f5353911266ce8f7f6524e7eb9921614ae78b7/src/Method/YarnMethod.php
>- https://github.com/factorial-io/phabalicious/commits/66f5353911266ce8f7f6524e7eb9921614ae78b7/src/Method/NpmMethod.php
>
> which would be the files that play a role in the switch from yarn to npm in phab.
>
> And there actually seem to be at least three commits that were only done for yarn but not npm:
>
>- https://github.com/factorial-io/phabalicious/commit/bede6793580ab13e71b190f155328d958de3d720
>- https://github.com/factorial-io/phabalicious/commit/a53c6248e2ec69d4619b9fe50ca04607bce1ed5d
>- https://github.com/factorial-io/phabalicious/commit/2a457c9f0de43bb67fbadb25203f0a52cbde4d3e

**Comment by stmh** [comment]
thanks for reporting. fixed in 3.8.25 and 4.0-dev

**Comment by stmh** [comment]
you can force the new version in your `.fabfile.yaml` -- add/update the following line:

```
requires: 3.8.35
```
then phab will exit early with a helpful error-message, when the version does not match.
